dhis2/dhis2-core
# refactor: changed the way relationship are identified inside preheat [DHIS2-10390]

Relationships are a particular kind of entity since they should be identified by type, from, and to rather than uids.
In this ticker, during preheat, a relationship-key for use in preheat map is generated considering relevant parameters and relationship types, including a bidirectional flag.

**Automatic testing**
E2E test provided.

**Manual testing**
let's consider the case that a relationship already exists between A and B (A-->B)

When posting an import payload with a relationship A-->B again it should not be duplicated in database, and import report should return that relationship as ignored

When posting an import payload with a relationship B-->A it should:
- if the relationship type is bidirectional, it should not be duplicated in the database, and the import report should return that relationship as ignored
- if the relationship type is not bidirectional, it should be created as B-->A in the database, and the import report should return that relationship as created
